+ source /root/mugen/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start to prepare the test environment.'
+ message='Start to prepare the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to prepare the test environment.'
Mon Jun 26 02:02:38 2023 - INFO  - Start to prepare the test environment.
+ DNF_INSTALL 'groovy18 tar'
+ pkgs='groovy18 tar'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen/libs/locallibs/rpm_manage.py install --pkgs 'groovy18 tar' --node 1 --tempfile ''
+ tmpfile2='Last metadata expiration check: 0:06:30 ago on Mon 26 Jun 2023 01:57:00 AM CST.
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: 
 Problem: package groovy18-1.8.9-1.oe2303.noarch requires mvn(org.codehaus.gpars:gpars), but none of the providers can be installed
  - package gpars-1.2.1-13.oe2303.noarch requires mvn(org.codehaus.groovy:groovy-all), but none of the providers can be installed
  - conflicting requests
  - nothing provides mvn(jline:jline) needed by groovy-2.4.8-11.oe2303.noarch
(try to add '\''--skip-broken'\'' to skip uninstallable packages or '\''--nobest'\'' to use not only best candidate packages)'
+ '[' -z '' ']'
+ tmpfile='Last metadata expiration check: 0:06:30 ago on Mon 26 Jun 2023 01:57:00 AM CST.
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: 
 Problem: package groovy18-1.8.9-1.oe2303.noarch requires mvn(org.codehaus.gpars:gpars), but none of the providers can be installed
  - package gpars-1.2.1-13.oe2303.noarch requires mvn(org.codehaus.groovy:groovy-all), but none of the providers can be installed
  - conflicting requests
  - nothing provides mvn(jline:jline) needed by groovy-2.4.8-11.oe2303.noarch
(try to add '\''--skip-broken'\'' to skip uninstallable packages or '\''--nobest'\'' to use not only best candidate packages)'
+ tar -xvf common/data.tar.gz
+ LOG_INFO 'End to prepare the test environment.'
+ message='End to prepare the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to prepare the test environment.'
Mon Jun 26 02:03:34 2023 - INFO  - End to prepare the test environment.
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run test.'
+ message='Start to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to run test.'
Mon Jun 26 02:03:36 2023 - INFO  - Start to run test.
+ grep -i '^groov' ./data/words
+ grep GROOVE
+ groovy18 -p -e 'line.toUpperCase()'
oe_test_groovy18_02.sh: line 31: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 -p failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -p failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 -p failed'
+ LOG_ERROR 'Check groovy18 -p failed'
+ message='Check groovy18 -p failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -p failed'
Mon Jun 26 02:03:37 2023 - ERROR - Check groovy18 -p failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 32'
+ message='oe_test_groovy18_02.sh line 32'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 32'
Mon Jun 26 02:03:39 2023 - ERROR - oe_test_groovy18_02.sh line 32
+ '[' 0 -eq 1 ']'
+ return 0
+ groovy18 -a -pe 'split[0].toInteger()+split[-2].toInteger()' ./data/accounts.txt
+ grep 5
oe_test_groovy18_02.sh: line 33: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 -a failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -a failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 -a failed'
+ LOG_ERROR 'Check groovy18 -a failed'
+ message='Check groovy18 -a failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -a failed'
Mon Jun 26 02:03:40 2023 - ERROR - Check groovy18 -a failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 34'
+ message='oe_test_groovy18_02.sh line 34'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 34'
Mon Jun 26 02:03:42 2023 - ERROR - oe_test_groovy18_02.sh line 34
+ '[' 0 -eq 1 ']'
+ return 0
+ find ./data -name 'test*.groovy'
+ xargs groovy18 -i ./data/Script.groovy -pe '(line =~ "//.*").replaceAll("")'
xargs: groovy18: No such file or directory
+ CHECK_RESULT 127 0 0 'Check groovy18 -i failed'
+ actual_result=127
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -i failed'
+ exit_mode=0
+ '[' -z 127 ']'
+ '[' 0 -eq 0 ']'
+ test 127x '!=' 0x
+ test -n 'Check groovy18 -i failed'
+ LOG_ERROR 'Check groovy18 -i failed'
+ message='Check groovy18 -i failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -i failed'
Mon Jun 26 02:03:43 2023 - ERROR - Check groovy18 -i failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 36'
+ message='oe_test_groovy18_02.sh line 36'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 36'
Mon Jun 26 02:03:45 2023 - ERROR - oe_test_groovy18_02.sh line 36
+ '[' 0 -eq 1 ']'
+ return 0
+ rm -f './data/*.bak'
+ grep -i '^groov' ./data/words
+ groovy18 -n -e 'println line.toUpperCase()'
+ grep GROOVE
oe_test_groovy18_02.sh: line 38: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 -n failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -n failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 -n failed'
+ LOG_ERROR 'Check groovy18 -n failed'
+ message='Check groovy18 -n failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -n failed'
Mon Jun 26 02:03:47 2023 - ERROR - Check groovy18 -n failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 39'
+ message='oe_test_groovy18_02.sh line 39'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 39'
Mon Jun 26 02:03:48 2023 - ERROR - oe_test_groovy18_02.sh line 39
+ '[' 0 -eq 1 ']'
+ return 0
+ groovy18 --debug -e 'throw new IOException()'
+ grep IOException trace.data
+ CHECK_RESULT 1 0 0 'Check groovy18 --debug failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 --debug failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 --debug failed'
+ LOG_ERROR 'Check groovy18 --debug failed'
+ message='Check groovy18 --debug failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 --debug failed'
Mon Jun 26 02:03:50 2023 - ERROR - Check groovy18 --debug failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 42'
+ message='oe_test_groovy18_02.sh line 42'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 42'
Mon Jun 26 02:03:51 2023 - ERROR - oe_test_groovy18_02.sh line 42
+ '[' 0 -eq 1 ']'
+ return 0
+ rm -f trace.data
+ groovy18 -d -e 'throw new IOException()'
+ grep IOException trace.data
+ CHECK_RESULT 1 0 0 'Check groovy18 -d failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -d failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 -d failed'
+ LOG_ERROR 'Check groovy18 -d failed'
+ message='Check groovy18 -d failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -d failed'
Mon Jun 26 02:03:53 2023 - ERROR - Check groovy18 -d failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 46'
+ message='oe_test_groovy18_02.sh line 46'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 46'
Mon Jun 26 02:03:54 2023 - ERROR - oe_test_groovy18_02.sh line 46
+ '[' 0 -eq 1 ']'
+ return 0
+ rm -f trace.data
+ groovy18 -D message=world -e 'println '\''Hello, '\'' + System.getProperty('\''message'\'')'
+ grep Hello
oe_test_groovy18_02.sh: line 48: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 -D failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -D failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 -D failed'
+ LOG_ERROR 'Check groovy18 -D failed'
+ message='Check groovy18 -D failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -D failed'
Mon Jun 26 02:03:56 2023 - ERROR - Check groovy18 -D failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 49'
+ message='oe_test_groovy18_02.sh line 49'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 49'
Mon Jun 26 02:03:57 2023 - ERROR - oe_test_groovy18_02.sh line 49
+ '[' 0 -eq 1 ']'
+ return 0
+ groovy18 --define message=world -e 'println '\''Hello, '\'' + System.getProperty('\''message'\'')'
+ grep Hello
oe_test_groovy18_02.sh: line 50: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 --define failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 --define failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 --define failed'
+ LOG_ERROR 'Check groovy18 --define failed'
+ message='Check groovy18 --define failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 --define failed'
Mon Jun 26 02:03:59 2023 - ERROR - Check groovy18 --define failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 51'
+ message='oe_test_groovy18_02.sh line 51'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 51'
Mon Jun 26 02:04:00 2023 - ERROR - oe_test_groovy18_02.sh line 51
+ '[' 0 -eq 1 ']'
+ return 0
+ groovy18 --encoding UTF-8 ./data/DataTypeTest.groovy
+ grep name:张三
oe_test_groovy18_02.sh: line 52: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 --encoding failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 --encoding failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 --encoding failed'
+ LOG_ERROR 'Check groovy18 --encoding failed'
+ message='Check groovy18 --encoding failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 --encoding failed'
Mon Jun 26 02:04:02 2023 - ERROR - Check groovy18 --encoding failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 53'
+ message='oe_test_groovy18_02.sh line 53'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 53'
Mon Jun 26 02:04:03 2023 - ERROR - oe_test_groovy18_02.sh line 53
+ '[' 0 -eq 1 ']'
+ return 0
+ groovy18 -cUTF-8 ./data/DataTypeTest.groovy
+ grep name:张三
oe_test_groovy18_02.sh: line 54: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 -c failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 -c failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 -c failed'
+ LOG_ERROR 'Check groovy18 -c failed'
+ message='Check groovy18 -c failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 -c failed'
Mon Jun 26 02:04:05 2023 - ERROR - Check groovy18 -c failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 55'
+ message='oe_test_groovy18_02.sh line 55'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 55'
Mon Jun 26 02:04:06 2023 - ERROR - oe_test_groovy18_02.sh line 55
+ '[' 0 -eq 1 ']'
+ return 0
+ groovy18 --disableopt int ./data/DataTypeTest.groovy
+ grep name:张三
oe_test_groovy18_02.sh: line 56: groovy18: command not found
+ CHECK_RESULT 1 0 0 'Check groovy18 --disableopt failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check groovy18 --disableopt failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check groovy18 --disableopt failed'
+ LOG_ERROR 'Check groovy18 --disableopt failed'
+ message='Check groovy18 --disableopt failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check groovy18 --disableopt failed'
Mon Jun 26 02:04:08 2023 - ERROR - Check groovy18 --disableopt failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_groovy18_02.sh line 57'
+ message='oe_test_groovy18_02.sh line 57'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_groovy18_02.sh line 57'
Mon Jun 26 02:04:09 2023 - ERROR - oe_test_groovy18_02.sh line 57
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'End to run test.'
+ message='End to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to run test.'
Mon Jun 26 02:04:11 2023 - INFO  - End to run test.
+ CASE_RESULT 0
+ case_re=0
+ test -z 11
+ test 11 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Mon Jun 26 02:04:12 2023 - ERROR - failed to execute the case.
+ exit 11
+ post_test
+ LOG_INFO 'Start to restore the test environment.'
+ message='Start to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to restore the test environment.'
Mon Jun 26 02:04:13 2023 - INFO  - Start to restore the test environment.
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z Last metadata expiration check: 0:06:30 ago on Mon 26 Jun 2023 01:57:00 AM CST.
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: 
 Problem: package groovy18-1.8.9-1.oe2303.noarch requires mvn(org.codehaus.gpars:gpars), but none of the providers can be installed
  - package gpars-1.2.1-13.oe2303.noarch requires mvn(org.codehaus.groovy:groovy-all), but none of the providers can be installed
  - conflicting requests
  - nothing provides mvn(jline:jline) needed by groovy-2.4.8-11.oe2303.noarch
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages) ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile 'Last metadata expiration check: 0:06:30 ago on Mon 26 Jun 2023 01:57:00 AM CST.
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: 
 Problem: package groovy18-1.8.9-1.oe2303.noarch requires mvn(org.codehaus.gpars:gpars), but none of the providers can be installed
  - package gpars-1.2.1-13.oe2303.noarch requires mvn(org.codehaus.groovy:groovy-all), but none of the providers can be installed
  - conflicting requests
  - nothing provides mvn(jline:jline) needed by groovy-2.4.8-11.oe2303.noarch
(try to add '\''--skip-broken'\'' to skip uninstallable packages or '\''--nobest'\'' to use not only best candidate packages)'
Traceback (most recent call last):
  File "/root/mugen/libs/locallibs/rpm_manage.py", line 181, in <module>
    exitcode = rpm_remove(args.node, args.pkgs, args.tempfile)
  File "/root/mugen/libs/locallibs/rpm_manage.py", line 152, in rpm_remove
    with open(tmpfile, "r") as f:
OSError: [Errno 36] File name too long: "Last metadata expiration check: 0:06:30 ago on Mon 26 Jun 2023 01:57:00 AM CST.\nPackage tar-2:1.34-4.oe2303.riscv64 is already installed.\nError: \n Problem: package groovy18-1.8.9-1.oe2303.noarch requires mvn(org.codehaus.gpars:gpars), but none of the providers can be installed\n  - package gpars-1.2.1-13.oe2303.noarch requires mvn(org.codehaus.groovy:groovy-all), but none of the providers can be installed\n  - conflicting requests\n  - nothing provides mvn(jline:jline) needed by groovy-2.4.8-11.oe2303.noarch\n(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)"
+ '[' 0 -ne 0 ']'
+ rm -rf ./data
+ LOG_INFO 'End to restore the test environment.'
+ message='End to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to restore the test environment.'
Mon Jun 26 02:04:21 2023 - INFO  - End to restore the test environment.
